src/SessionMiddleware.php

```diff
@@ -21,15 +21,21 @@ class SessionMiddleware implements MiddlewareInterface
      */
     private $persistence;
 
-    public function __construct(SessionPersistenceInterface $persistence)
+    /**
+     * @var SessionInterface
+     */
+    private $session;
+
+    public function __construct(SessionPersistenceInterface $persistence, SessionInterface $session)
     {
         $this->persistence = $persistence;
+        $this->session = $session;
     }
 
     public function process(ServerRequestInterface $request, DelegateInterface $delegate) : ResponseInterface
     {
-        $session = new LazySession($this->persistence, $request);
-        $response = $delegate->process($request->withAttribute(self::SESSION_ATTRIBUTE, $session));
+        $this->session->setRequest($request);
```

Comment:
Since `setRequest()` is not part of `SessionInterface`, this particular call is problematic; what if a `LazySession` is not injected, but some other implementation instead?

Reply:
ie a good catch 👍 . We may want to see if the object is `LazySession` itself. Too much of complications now :) .